src/api/c/convolve.cpp

```diff
@@ -74,6 +74,8 @@ af_err convolve(af_array *out, const af_array signal, const af_array filter)
 
         AF_BATCH_KIND convBT = identifyBatchKind<baseDim>(sdims, fdims);
 
+        ARG_ASSERT(1, (sdims.ndims()>=1));
+        ARG_ASSERT(2, (fdims.ndims()>=1));
```

Comment:
We should just check `std::min(sdims.ndims(), fdims.ndims())` and put to the right fft.